### ISSUE TITLE
Add "Lock Wallet" service to the UI when password-protection has been enabled

### ISSUE THREAD

**Issue by dafacto** [BlueWallet/BlueWallet#1317]
When the two-password plausible deniability feature is in use, the only method of switching between the two sets of wallets is to manually kill the app. This is problematic for a number of reasons, in order of increasing importance:

- Killing an app should never be part of a UI workflow, as it's not discoverable, and many people are unaware how to even do that.

- If leaves the app, and happens to forget to kill it, the last active wallet set will be visible on the next launch. This compromises the whole point of the plausible deniability feature.

The proposed solution is:

- When password protection is enabled, provide a Lock icon in the UI, which locks the active set of wallets.

- Add an unlock screen where the user enters a password, replacing the modal prompt.

The presence of the Lock icon does not compromise plausible deniability, since a second password is only an optional extension of the basic password protection, applicable to a single set of wallets.

**Comment by marcosrdz** [comment]
@Overtorment  @ncoelho  what do u think?

**Comment by Overtorment** [comment]
yeah, plausible deniability mostly relies on OS that is supposed to terminate the app when its not active for a long time or when there is RAM shortage.

maybe we can put `secure logout` somewhere on `settings` screen (so it wont obstruct view on other screens. view space on mobile apps is very precious). looks like the easiest solution

**Comment by dafacto** [comment]
Having thought about it, I would label it “Lock Wallet”, rather than “Logout”.

**Comment by ncoelho** [comment]
ack for a settings UI element

**Comment by marcosrdz** [comment]
https://github.com/BlueWallet/BlueWallet/pull/1362

**Comment by dafacto** [comment]
A summary of my view is:

1. The risk is that if they leave the app unlocked, and are shortly thereafter forced to re-open it, the last open wallet is exposed, defeating the purpose of plausible deniability. 
1. The UI must indicate to the user that the app is unlocked, and provide a means of unlocking it. We can't expect the user (who frequently interact with many many apps) to remember that they must force-quit Blue Wallet each time they use it. Shake-to-lock is a non-starter, since whatever in the UI would indicate that needs to be done, could itself simply be a tap-based control.
1. Whatever UI control is added (I suggest a lock icon), the current modal unlock prompt needs to be replaced with a lock screen, containing a password field (with natural escape options like "recovery wallet", etc.) This lock screen would be present on launch, and after the user manually locks the wallet.

**Comment by ncoelho** [comment]
When solving a problem it is important to look to the root cause of the problem and not focus on a specific prescribed solution, but look into different solutions. The problem we are solving is:

- The wallet is unlocked when the app goes into the background, which can make the plausible deniability not useful.

Trying to solve this can generate more problems, which need to be taken in consideration as well:
- Like the user has to enter the password all the time for instance, which is not ideal.
- A UI element is not ideal as well, it will take space away for a non primary function. 

Are there other solutions that could mitigate this, for instance. 
- Time? Can we destroy the lock every 1h (x time...) after the app was last used
- Can we prompt the user before going in background?
- Are there other visual clues, is there anyone doing a similar solution?

I would like to see solutions/ideas explored, instead of trying to prescribe a solution without any data, feedback or context, which is not very useful in my view.

**Comment by dafacto** [comment]
I agree with your base view that it's better to discuss problems than prescribe solutions. The reason I'm prescribing a solution here is that the company I own (makaluinc.com) has been developing (some very successful) apps for many years, and experience sometimes allows shortcuts in discussions. Not trying to appeal to authority or anything; just explain why I'm jumping to solutions.

As far as I know, Blue Wallet is the only mobile wallet that provides this feature, and can be a market differentiator for you (if you want). But the implementation needs to be far better than it is now.

A user who elects to enable plausible deniability, is telling you that they want to keep significant value in the wallet. They are telling you that they plan to use Blue Wallet for more than their spending money. Knowing this, the design trade-offs should prioritize minimizing user errors (exposing them to large loss) over inconvenience.

If a user plans to store $1M in their wallet, I can guarantee you that having to enter a password each time, or the presence of a prominent UI element, will NOT be a problem for them, if any alternative exposes them to the risk of messing up. You don't see any Ledger users complaining that they have to enter a PIN every time they start the device, and 5 minutes after its idle.

In my opinion, characteristics of any solution:

1. There must be a way for the user to manually lock the app. This then excludes, as the primary approach, things like time-outs.
1. For someone who has enabled plausible deniability, the unlocked state of the app needs to be clear to them in the UI, to minimize the chance they leave the app without unlocking. For this user, they will prefer that priority is given to this, as the possibility of making a mistake and leaving the wallet open is something they'll very much want to avoid.
1. Locking the app must take the app to a lock screen. This suggest removing the current modal prompt. Additionally, a lock screen provides the opportunity to provide fall-back options in case a user doesn't remember their password (wallet recovery, how to contact someone, etc.)
